lib/simp/cli/utils.rb

```diff
@@ -98,31 +137,47 @@ def generate_password(length = DEFAULT_PASSWORD_LENGTH )
 
   # Add custom facts to the Facter search path and then reload the facts
   #
-  # +module_paths+: Array of module paths in which to find custom facts
-  # +add_to_env+:   Whether to add the custom fact paths to the FACTERLIB
-  #   environment variable, in addition, so that the facts available to
-  #   any spawned processes that use FACTERLIB, (e.g. `puppet apply`)
-  def load_custom_facts(module_paths, add_to_env = false)
+  # This can be called as many times as necessary and will re-load the facts
+  # each time with the new path.
+  #
+  # @param module_paths [Array[String]]
+  #   Paths to search for custom facts
+  # @param add_to_env [Boolean]
+  #   Whether to add the custom fact paths to the FACTERLIB environment
+  #   variable, in addition, so that the facts available to any spawned
+  #   processes that use FACTERLIB, (e.g. `puppet apply`)
+  def load_custom_facts(module_paths=[], add_to_env = false)
+    # Missing directories do not matter since they will be skipped
+    default_module_paths = [
+      '/opt/puppetlabs/server/data/environments',
+      '/usr/share/simp/modules'
+    ].map{|x| File.absolute_path(x)}
+
     fact_paths = []
-    Facter.clear  # Facter.loadfacts won't reload without this
-    module_paths.each do |dir|
+    Facter.clear # Facter.loadfacts won't reload without this
+
+    # Later facts override earlier ones so the default paths are fine
+    (default_module_paths + Array(module_paths)).uniq.each do |dir|
       next unless File.directory?(dir)
       Find.find(dir) do |mod_path|
-        fact_path = File.expand_path('lib/facter', mod_path)
-        if File.directory?(fact_path)
-          Facter.search(fact_path)
-          fact_paths << fact_path
+        Find.prune unless File.directory?(mod_path)
+        if mod_path.end_with?('/lib/facter')
+          Facter.search(mod_path)
+          fact_paths << mod_path
         end
-        Find.prune unless mod_path == dir
       end
     end
+
+    # Called to get all of the facts that are included inside of core Facter
+    # for Puppet
+    Puppet.initialize_facts if Puppet.respond_to?(:initialize_facts)
```

Comment:
I'm honestly not sure. I found that if I didn't do this, I didn't pick up the facts that were native to Puppet when running `simp config` in Puppet 6. It may be different but this method has been present for a LONG time and I verified that it was in all the way back to Puppet 4 so there's no harm in leaving it. In fact, it's how Puppet itself primes its facts (which is where I found the call).

Reply:
Note: In `rspec`, you'll always get the facts because it only uses Facter 2.X